eio/tls_eio.ml

```diff
@@ -22,14 +22,17 @@ module Raw = struct
     | End_of_file as ex ->
       t.state <- `Eof;
       raise ex
+    | Eio.Cancel.Cancelled _ as ex ->
+      (* No need to break the flow just because a read got cancelled. *)
+      raise ex
     | exn ->
```

Comment:
Wouldn't it make sense to completely remove this bit instead? I believe this is trapping other exceptions in addition to `Cancelled`, such as 'Poisioned' and intermittently this https://github.com/ocaml/ocaml/blob/trunk/stdlib/effect.ml#L59